File: android/src/main/java/com/backgroundlocation/event/ConnectivityChangeEvent.java
package com.backgroundlocation.event;

import org.json.JSONObject;

/**
 * Connectivity Change Event
 * Type-safe event class
 */
public class ConnectivityChangeEvent {
    private final boolean connected;
    
    public ConnectivityChangeEvent(boolean connected) {
        this.connected = connected;
    }
    
    public boolean isConnected() {
        return connected;
    }
    
    public JSONObject toJson() {
        try {
            JSONObject json = new JSONObject();
            json.put("connected", connected);
            return json;
        } catch (Exception e) {
            return new JSONObject();
        }
    }
    
    public String getEventName() {
        return "connectivitychange";
    }
}

File: ios/event/ConnectivityChangeEvent.m
//
//  ConnectivityChangeEvent.m
//  RNBackgroundLocation
//
//  Connectivity Change Event
//  Android ConnectivityChangeEvent.java benzeri
//

#import "ConnectivityChangeEvent.h"

@implementation ConnectivityChangeEvent

- (instancetype)initWithConnected:(BOOL)connected {
    self = [super init];
    if (self) {
        _connected = connected;
    }
    return self;
}

- (NSDictionary *)toDictionary {
    return @{
        @"connected": @(self.connected)
    };
}

- (NSString *)getEventName {
    return @"connectivitychange";
}

@end





